### PR TITLE
Use group attribute for KV version and add DB engine support

### PR DESCRIPTION
**Pull request by bimtauer**
# TL;DR
Removes the KV Version config parameter and instead uses the Secret `group` field to allow using both (and other Vault secret backends) dynamically. 
This also allows using the [Database Secrets Engine](https://developer.hashicorp.com/vault/docs/secrets/databases), which effectively means noop since we want to read the whole credential and not just a specific key.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
[The Secret protobuf defines a group field](https://github.com/flyteorg/flyteidl/blob/master/protos/flyteidl/core/security.proto#L33) which we have previously ignored. Leveraging that field allows to simplify the secret manager config and allows supporting multiple Vault secret backends dynamically by deciding what secret template to inject based on the optional `group` parameter.

This is a breaking change for people using the previous version of the Vault Secret Manager. To migrate they will need to remove the `kvVersion` key from their propeller configs and specify the appropriate group version in their secret requests, like for example so:

```python
@task(
    secret_requests=[
        Secret(group="foo", key="bar", group_version="kv2"),
    ]
)
```

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/webhook/config/config.go [hunk 1]
```diff
@@ -90,8 +78,7 @@ type AWSSecretManagerConfig struct {
 }
 
 type VaultSecretManagerConfig struct {
-	Role      string    `json:"role" pflag:",Specifies the vault role to use"`
-	KVVersion KVVersion `json:"kvVersion" pflag:"-,The KV Engine Version. Defaults to 2. Use 1 for unversioned secrets. Refer to - https://www.vaultproject.io/docs/secrets/kv#kv-secrets-engine."`
```
[the remark this answers is not in the document]
Comment (hamersaw) [reply pkg/webhook/config/config.go:94]: @bimtauer soft ping

[the remark this answers is not in the document]
Comment (bimtauer) [reply pkg/webhook/config/config.go]: Hi there, I just pushed an update that preserves the current config and keeps its setting an overwritable default. Wdyt? Can I somehow add some user facing annotation to make sure that we want to deprecate that KVVersion config field eventually?

[the remark this answers is not in the document]
Comment (bimtauer) [reply pkg/webhook/config/config.go]: Rebased onto recent changes. I did also make the new test cases a bit more descriptive.

[the remark this answers is not in the document]
Comment (bimtauer) [reply pkg/webhook/config/config.go]: Looks like your CI doesn't trigger :/

**Comment by hamersaw** [comment]
@bimtauer looks great, thanks! Figuring out the CI trigger and once checks pass I think we can merge this.

I'm sure you're intimately more familiar with the vault docs than I, but it seems like we should update [similar to this change](https://github.com/flyteorg/flytesnacks/pull/986) as well?

**Comment by hamersaw** [comment]
@bimtauer  it seems like could be a [github issue](https://www.githubstatus.com/incidents/pr3498h3qkfy), any chance you can push a pseudo commit to retrigger to checks? I can't seem to find how to manually fire these off.

**Comment by bimtauer** [comment]
> @bimtauer looks great, thanks! Figuring out the CI trigger and once checks pass I think we can merge this.
> 
> I'm sure you're intimately more familiar with the vault docs than I, but it seems like we should update [similar to this change](https://github.com/flyteorg/flytesnacks/pull/986) as well?

Indeed, we should. I will prepare a second PR

**Comment by bimtauer** [comment]
Contributing to flytesnacks cookbook I noticed that docs development is quite cumbersome. I had to install torch and cublas as part of `docs-requirements.txt` just to see if my changes render nicely. Maybe that could be made easier.
